modules/frontend/src/main/scala/latis/logviz/EventComponent.scala

```diff
@@ -60,8 +61,18 @@ class EventComponent(
       canvas      =  canvasIO.asInstanceOf[HTMLCanvasElement]
       context     =  canvas.getContext("2d").asInstanceOf[dom.CanvasRenderingContext2D]
       eParser     <- Resource.eval(EventParser())
-      _           <- Resource.eval(sup.supervise(stream.evalTap(event =>
-                      eParser.parse(event).handleErrorWith(IO.println)).compile.drain).void)
+      parserRef   <- Resource.eval(Ref[IO].of(eParser))
+      _           <- Resource.eval(sup.supervise(signal.discrete.switchMap { stream => 
+                      // making a new parser
+                      val newParser = for {
+                        parser <- EventParser()
+                        _      <- parserRef.set(parser)
+                      } yield parser
+                      
+                      Stream.eval(newParser).flatMap { parser =>
+                        stream.evalTap(parser.parse)
```

Comment:
Need to preserve the error handling Jerry added:

```suggestion
                        stream.evalTap(
                          parser.parse(_).handleErrorWith(IO.println)
                        )
```